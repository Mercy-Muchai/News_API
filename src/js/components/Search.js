import React from 'react';
import ReactDOM from 'react-dom';
import axios from 'axios';

export default class Search extends React.Component {
  constructor() {
    super();
    this.state = {
      sources: [],
      search: 'Search Source'
    };
  }
  updateSearch(event) {
    this.setState({ search: event.target.value });

  }

  componentDidMount() {
    axios.get(`https://newsapi.org/v1/sources?language=en`)
      .then((result) => {
        this.setState({
          sources: result.data.sources
        });
      });
  }

  render() {
    let found = this.state.sources.filter(
      (sources) => {
        return sources.name.toLowerCase().indexOf(this.state.search.toLowerCase()) !== -1;

      });

    return (
      <div className="col-sm-12" classID="search" style={{ paddingTop: "25px" }}>

        <input type="text" placeholder='Search Source' value={this.state.search} onChange={this.updateSearch.bind(this)} style={{ backgroundColor: '#ffffff' }} className="searchbox" />
        <span id="tweet-container-index" style={{ paddingLeft: '1010px'}}>
          <a href="https://twitter.com/share" className="twitter-share-button" data-url={'https://getnews-api.herokuapp.com/#/abc-news-au/top'} data-text={'Get News here from all over the world'}>Tweet</a>
        </span>
        {found.map((sources) => {
          return (
            <div className="search" key={sources.id}>
              <a href={sources.url}> {sources.name} </a>
            </div>);
        }
        )}

      </div>
    );
  }
};