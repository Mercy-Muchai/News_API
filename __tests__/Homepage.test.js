import React from 'react';
import { shallow, wrapper } from 'enzyme';
import shallowToJSON from 'enzyme-to-json';
import App from '../src/js/components/app.jsx';
import Search from '../src/js/components/Search.jsx';

describe('Test Sources component', () => {
  const wrapper = shallow(<App />);
  it('should have a div', () => {
    expect(wrapper.node.type).toBe('div');
  });
});
// describe ('Search button', ()=> {
//  it('should render a search button', () => {
//     const wrapper = shallow(<App />);
//     expect(wrapper.containsMatchingElement(<Search />)).toBe(true);
//   });
// });

describe('Home Component', () => {
  it('Should render a home page', () => {
    const homes = shallow(<App />);
    const tree = shallowToJSON();
    expect(tree).toMatchSnapshot();
  });
});
