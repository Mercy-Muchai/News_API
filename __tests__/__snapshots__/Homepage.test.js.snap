// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Should render a home page apperance 1`] = `<div />`;
